code/modules/xenoarchaeology/components/xenoartifact.dm

```diff
@@ -0,0 +1,411 @@
+/*
+	Artifact component
+*/
+
+/datum/component/xenoartifact
+	///List of artifact-traits we have : list(PRIORITY = list(trait))
+	var/list/artifact_traits = list()
```

Comment:
This should be renamed to `trait_categories` or something
when you name it as `traits`, it means the list will have `list(TRAIT, TRAIT)`, but in fact, it was assoc list, and traits were stored inside the list of the list
the below one `blacklisted_traits` is okay